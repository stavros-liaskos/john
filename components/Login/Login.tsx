import React from 'react';
import { LoginProps } from './Login.types';

const Login: React.FunctionComponent<LoginProps> = ({ i18n, handleRegister }) => {
  if (!i18n?.loginBtn || !i18n.text || !i18n.welcome || !i18n.artistsCount || !i18n.releasesCount) {
    return null;
  }

  return (
    <div className="flex flex-col items-center justify-center lg:w-1/2">
      <h2 className="rr-text text-4xl mb-3">{i18n.welcome}</h2>
      <p className="rr-text text-xl mb-3">{i18n.text}</p>

      <button className="btn btn-large lg:ml-8 my-8 w-44 h-14" onClick={() => handleRegister()}>
        {i18n.loginBtn}
      </button>
    </div>
  );
};

export default Login;
