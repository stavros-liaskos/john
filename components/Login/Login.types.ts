type LoginI18n = {
  welcome: string;
  loginBtn: string;
  text: string;
  artistsCount: string;
  releasesCount: string;
};

export interface LoginProps {
  i18n: LoginI18n;
  handleRegister: Function;
}
