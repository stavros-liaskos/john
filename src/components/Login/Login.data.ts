export const loginI18n = {
  welcome: 'Welcome to Release Raccoon!',
  loginBtn: 'Register',
  text: "Receive your favorite artists' music in your email every week!",
  artistsCount: 'Artists',
  releasesCount: 'Releases',
};
