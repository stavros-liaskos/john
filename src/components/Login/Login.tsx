import React from 'react';
import { LoginProps } from './Login.types';
import Link from 'next/link';

const Login: React.FunctionComponent<LoginProps> = ({ i18n }) => {
  if (!i18n?.loginBtn || !i18n.text || !i18n.welcome || !i18n.artistsCount || !i18n.releasesCount) {
    return null;
  }

  return (
    <div className="flex flex-auto flex-col items-center justify-center w-full">
      <h2 className="rr-text text-center text-4xl mb-3">{i18n.welcome}</h2>
      <p className="rr-text text-center text-xl mb-3">{i18n.text}</p>

      <Link href="https://api.releaseraccoon.online/me?redirectUrl=https://www.releaseraccoon.online">
        <button className="btn btn-large lg:ml-8 mt-8 mb-24 w-44 h-14">{i18n.loginBtn}</button>
      </Link>

      <Link href="/api/auth/login">
        <button className="btn btn-large lg:ml-8 mt-8 mb-24 w-44 h-14">2. Take me to dashboard</button>
      </Link>
    </div>
  );
};
Login.whyDidYouRender = true;
export default Login;
