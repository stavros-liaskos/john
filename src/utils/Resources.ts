import { components } from '../types/schema';

interface IResources {
  cacheMap: Record<string, unknown>;
  fetch: (url: string) => void;
}

class Resources implements IResources {
  private cacheMap;

  constructor() {
    this.cacheMap = new Map();
  }

  fetch(url) {
    if (this.cacheMap.has(url)) {
      return this.cacheMap.get(url);
    }

    const promise = fetch(url, {
      method: 'GET',
      credentials: 'include',
    })
      .then(response => {
        return response.json();
      })
      .then((data: components['schemas']['FollowedArtistsResponse']) => {
        this.cacheMap.set(url, data);
        return data;
      });

    throw promise;
  }
}

const resources = new Resources();

export default resources;
