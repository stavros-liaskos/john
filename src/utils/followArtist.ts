import { components } from '../types/schema';
import Endpoints from '../types/endpoints';

async function followArtist(artist: components['schemas']['SearchResultArtistDto'], cb: () => void) {
  const headers = new Headers({ 'Content-Type': 'application/json' });
  await fetch(Endpoints.Follow, {
    method: 'POST',
    credentials: 'include',
    headers,
    body: JSON.stringify(artist),
  })
    .then(() => {
      console.log(`${artist.name} followed successfully`);
    })
    .catch(error => {
      console.error('Error:', JSON.stringify(error));
    })
    .finally(() => {
      cb();
    });
}
export default followArtist;
