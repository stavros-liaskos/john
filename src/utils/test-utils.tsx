import React from 'react';
import { render, act } from '@testing-library/react';
import ThemeProvider from '../contexts/Theme/ThemeProvider';
import { UserProvider } from '@auth0/nextjs-auth0/client';
import ArtistsListProvider from '../contexts/ArtistsList/ArtistsListProvider';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <ThemeProvider>
      <UserProvider>
        <ArtistsListProvider>{children}</ArtistsListProvider>
      </UserProvider>
    </ThemeProvider>
  );
};

const customRender = (ui: React.ReactElement, options?: { [key: string]: unknown }) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };

export const renderWithAct = async (ui: React.ReactElement, options?: { [key: string]: unknown }) => {
  const component = render(ui, { wrapper: AllTheProviders, ...options });

  const promise = Promise.resolve();
  await act(() => promise);

  return component;
};

export const beforeEachTest = () => {
  jest.resetModules();
  jest.clearAllMocks();
  jest.restoreAllMocks();
  localStorage.clear();
};
