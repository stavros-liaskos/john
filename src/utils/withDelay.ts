export default function withDelay(cb: () => void, delay = 1000) {
  return setTimeout(() => {
    cb();
  }, delay);
}
