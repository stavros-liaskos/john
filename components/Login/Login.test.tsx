import React from 'react';
import { fireEvent, renderWithAct } from '../../utils/test-utils';
import Login from './Login';

const props = {
  i18n: {
    welcome: 'Welcome to Release Raccoon!',
    loginBtn: 'Register',
    text: "Receive your favorite artists' music in your email every week!",
    artistsCount: 'Artists',
    releasesCount: 'Releases',
  },
  handleRegister: () => {},
  counters: {
    artistsCounter: 4965,
    releasesCounter: 3816,
  },
};

describe('Login', () => {
  it('renders without data without crashing', async () => {
    // @ts-ignore
    await renderWithAct(<Login />);
  });

  it('renders with data without crashing', async () => {
    const { container } = await renderWithAct(<Login {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('redirects when Register btn is clicked', async () => {
    const handleRegister = jest.fn();
    const { getByText } = await renderWithAct(<Login {...props} handleRegister={handleRegister} />);

    fireEvent.click(getByText(/Register/));
    expect(handleRegister).toHaveBeenCalledTimes(1);
  });
});
