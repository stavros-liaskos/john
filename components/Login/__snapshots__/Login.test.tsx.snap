// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders with data without crashing 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center lg:w-1/2"
  >
    <h2
      class="rr-text text-4xl mb-3"
    >
      Welcome to Release Raccoon!
    </h2>
    <p
      class="rr-text text-xl mb-3"
    >
      Receive your favorite artists' music in your email every week!
    </p>
    <button
      class="btn btn-large lg:ml-8 my-8 w-44 h-14"
    >
      Register
    </button>
    <div
      class="flex justify-between w-full px-16 mt-16"
    >
      <div
        class="text-center"
      >
        <p
          class="rr-text"
        >
          Artists
        </p>
        <p
          class="rr-text"
        >
          4965
        </p>
      </div>
      <div
        class="text-center"
      >
        <p
          class="rr-text"
        >
          Releases
        </p>
        <p
          class="rr-text"
        >
          3816
        </p>
      </div>
    </div>
  </div>
</div>
`;
